src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Initializers/WebJobsTelemetryInitializer.cs

```diff
@@ -43,21 +43,12 @@ public void Initialize(ITelemetry telemetry)
             }
 
             telemetry.Context.Cloud.RoleInstance = _roleInstanceName;
-
-            RequestTelemetry request = telemetry as RequestTelemetry;
-
-            // Zero out all IP addresses other than Requests
-            if (request == null)
+            if (telemetry.Context.Location.Ip == null)
             {
                 telemetry.Context.Location.Ip = LoggingConstants.ZeroIpAddress;
```

Comment:
@brettsam I was also wondering if we can drop this? I don't remember what's the reason for the overriding null IP with 0.0.0.0 ?

Reply:
I also don't remember why we did that, unfortunately. At one point, was it required? But I'd rather not risk removing it -- just in case someone depends on it.